PR: https://github.com/odoo/odoo/pull/35716

From: 2913337fd7addc8400606e98100a81a7d2af0fc9
From: Joseph Caburnay
Date: 2019-08-19 07:27:13

Structural Changes: 5
Total Changes: 7

[IMP] point_of_sale: B2B ease navigation within PoS

TASK-ID: 2050245

closes odoo/odoo#35716

Signed-off-by: pimodoo <pimodoo@users.noreply.github.com>

================================= pseudo patch: =================================

--- a/addons/point_of_sale/models/pos_category.py
+++ b/addons/point_of_sale/models/pos_category.py
@@ -13,7 +13,7 @@ class PosCategory(models.Model):
         if not self._check_recursion():
             raise ValueError(_('Error ! You cannot create recursive categories.'))
 
-    name = fields.Char(required=True, translate=True)
+    name = fields.Char(string='Category Name', required=True, translate=True)
     parent_id = fields.Many2one('pos.category', string='Parent Category', index=True)
     child_id = fields.One2many('pos.category', 'parent_id', string='Children Categories')
     sequence = fields.Integer(help="Gives the sequence order when displaying a list of product categories.")

--- a/addons/point_of_sale/models/pos_config.py
+++ b/addons/point_of_sale/models/pos_config.py
@@ -202,6 +202,7 @@ class PosConfig(models.Model):
              "the closing of his session saying that he needs to contact his manager.")
     payment_method_ids = fields.Many2many('pos.payment.method', string='Payment Methods', default=lambda self: self._default_payment_methods())
     company_has_template = fields.Boolean(string="Company has chart of accounts", compute="_compute_company_has_template")
+    current_user_id = fields.Many2one('res.users', string='Current Session Responsible', compute='_compute_current_session_user')
 
     @api.depends('company_id')
     def _compute_company_has_template(self):
@@ -262,10 +263,12 @@ class PosConfig(models.Model):
                 pos_config.pos_session_duration = (
                     datetime.now() - session[0].start_at
                 ).days if session[0].start_at else 0
+                pos_config.current_user_id = session[0].user_id
             else:
                 pos_config.pos_session_username = False
                 pos_config.pos_session_state = False
                 pos_config.pos_session_duration = 0
+                pos_config.current_user_id = False
 
     @api.constrains('cash_control')
     def _check_session_state(self):

--- a/addons/point_of_sale/models/pos_order.py
+++ b/addons/point_of_sale/models/pos_order.py
@@ -186,7 +186,7 @@ class PosOrder(models.Model):
     )
     note = fields.Text(string='Internal Notes')
     nb_print = fields.Integer(string='Number of Print', readonly=True, copy=False, default=0)
-    pos_reference = fields.Char(string='PoS Order Reference', readonly=True, copy=False)
+    pos_reference = fields.Char(string='Receipt Number', readonly=True, copy=False)
     sale_journal = fields.Many2one('account.journal', related='session_id.config_id.journal_id', string='Sales Journal', store=True, readonly=True)
     fiscal_position_id = fields.Many2one(
         comodel_name='account.fiscal.position', string='Fiscal Position',
